data/data/baremetal/masters/main.tf

```diff
@@ -1,3 +1,11 @@
+# This null resource is used to wait for the Ironic API to become available before trying to create
+# the master hosts.
+resource "null_resource" "api-wait" {
+  provisioner "local-exec" {
+    command = "timeout 900 bash -c 'while [[ \"$(curl -s -o /dev/null -w \"%%{http_code}\" ${var.ironic_uri})\" != \"200\" ]]; do sleep 5; done'"
+  }
+}
+
```

Comment:
I wonder if gophercloud would accept a generic function that waits for a specified time until an endpoint responds?

Or there could be a check on https://developer.openstack.org/api-ref/baremetal/#list-api-versions which I think is common to all OpenStack APIs?

This seems fine for a first step though.

Reply:
Also there is https://www.terraform.io/docs/providers/http/data_source.html but that doesn't have a timeout option...

Reply:
I can't imagine gophercloud taking something like that. I could change the terraform provider to support retrying if this is too ugly, but it felt more natural to me and explicit about what's going on  to handle this using terraform HCL.